themes/default/content/blog/aws-proton-pulumi/index.md

```diff
@@ -0,0 +1,171 @@
+
+## What are AWS Proton and AWS CodeBuild?
+
+AWS Proton is a service that enables platform engineers to define, vend, and maintain self-service infrastructure templates for deploying cloud resources in an automated workflow. Development and operations teams can leverage these templates to deploy well-architected, modern infrastructure  with confidence. Platform teams can create Proton templates for [environments](https://docs.aws.amazon.com/proton/latest/userguide/ag-environments.html) (which typically include shared resources like VPCs and container scheduler clusters) and application [services](https://docs.aws.amazon.com/proton/latest/userguide/ag-services.html) which utilize the resources in the environment. Templates may be stored in git repositories for easy version control and automatic syncing with the Proton service.
+
+AWS CodeBuild is a managed continuous integration (CI) service. Using CodeBuild, engineers can specify a Docker build container image, either [pre-built](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html) or [custom](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-docker-custom-image.html), and a list of commands to execute in the container to perform the build. AWS handles concerns like creating and destroying the container, logging, and metrics. When using CodeBuild as a provisioner for AWS Proton, teams specify the steps to both provision and tear down our infrastructure templates.
+
+## Using AWS Proton with CodeBuild and Pulumi
+
+Since CodeBuild provisioning in Proton allows us to execute arbitrary commands, we can script our template builds using the [Pulumi CLI](https://www.pulumi.com/docs/reference/cli/). AWS has built a complete, end-to-end example that can be found here: [https://github.com/aws-containers/proton-codebuild-provisioning-examples/tree/main/pulumi](https://github.com/aws-containers/proton-codebuild-provisioning-examples/tree/main/pulumi), and  for this post we are going to focus on the specific steps required to use CodeBuild provisioning in AWS Proton with Pulumi. Note that in this example we are still using the Pulumi Service as our backend, so we can still enjoy all the benefits the service provides like state file management, secure secrets handling, a shared history of changes to our stacks, and [advanced features available in Enterprise and Business Critical editions](https://www.pulumi.com/enterprise/).
```

Comment:
```suggestion
Since CodeBuild provisioning in Proton allows us to execute arbitrary commands, we can script our template builds using the [Pulumi CLI](https://www.pulumi.com/docs/reference/cli/). AWS has built a complete, end-to-end example that can be found [at GitHub](https://github.com/aws-containers/proton-codebuild-provisioning-examples/tree/main/pulumi), and  for this post we are going to focus on the specific steps required to use CodeBuild provisioning in AWS Proton with Pulumi. Note that in this example we are still using the Pulumi Service as our backend, so we can still enjoy all the benefits the service provides like state file management, secure secrets handling, a shared history of changes to our stacks, and [advanced features available in Enterprise and Business Critical editions](https://www.pulumi.com/enterprise/).
```

Reply:
Bit of a nit, but we shouldn't fully-qualify links like these; we should make them root-relative:

```
[Pulumi CLI](/docs/reference/cli/)
``` 

We can lint for this also, just haven't yet.

Reply:
```suggestion
Since CodeBuild provisioning in Proton allows us to execute arbitrary commands, we can script our template builds using the [Pulumi CLI](/docs/reference/cli/). AWS has built a complete, end-to-end example that can be found here: [https://github.com/aws-containers/proton-codebuild-provisioning-examples/tree/main/pulumi](https://github.com/aws-containers/proton-codebuild-provisioning-examples/tree/main/pulumi), and  for this post we are going to focus on the specific steps required to use CodeBuild provisioning in AWS Proton with Pulumi. Note that in this example we are still using the Pulumi Service as our backend, so we can still enjoy all the benefits the service provides like state file management, secure secrets handling, a shared history of changes to our stacks, and [advanced features available in Enterprise and Business Critical editions](/enterprise/).
```